src/module.c

```diff
@@ -13489,6 +13492,96 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
     return ctx->dbid;
 }
 
+/* --------------------------------------------------------------------------
+ * ## Key API for Set type
+ *
+ * See also RM_ValueLength(), which returns the cardinality of a set.
+ * -------------------------------------------------------------------------- */
+
+/* Operate a set key.
+ * REDISMODULE_SET_ADD: Add new elements into a set. If the key is an empty key
+ *                      open for writing, it is created with an empty set key in
+ *                      order to add the elements.
+ * REDISMODULE_SET_REM: Remove the specified element from the set and the key will be
+ *                      removed if has no any element in after remove elements operation.
+ * REDISMODULE_SET_ISMEMBER: Determines if member is a member of the set stored at key .
+ *
+ * * **key**: The set key to be operated.
+ * * **op**: The operation to the set key.
+ * * **flags**: The flag for following arguments. 
+ *      REDISMODULE_SET_CSTRING: The following arguments are null terminated C
+ *                               strings instead of RedisModuleString objects.
+ * * **...**: The actual arguments.
+ *
+ * Return value:
+ *
+ * REDISMODULE_SET_ADD: The number of elements were really added.
+ * REDISMODULE_SET_REM: The number of elements were really removed.
+ * REDISMODULE_SET_ISMEMBER: 1 if all the elements are member of the set. Otherwise return 0.
+ *
+ * If the return value is zero, `errno` is set as follows:
```

Comment:
Why can't we return -1 on error?

Reply:
Other APIs use ENOTSUP if the key doesn't exist or is of wrong type. (ENOENT is sometimes used if an element or entry within a key does not exist.)

Some APIs return -1 and set errno to ENOTSUP, for example [RedisModule_StreamTrimByID](https://redis.io/docs/reference/modules/modules-api-ref/#RedisModule_StreamTrimByID).